File: mobile-platform/ios/NoiceStreamView/NoiceStreamViewManager.m
//
//  NoiceStreamViewModule.m
//  MobilePlatform
//
//  Created by Tobias Helsing on 24.9.2024.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(NoiceStreamViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(onHeaderLayoutChanged, RCTBubblingEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onContentLayoutChanged, RCTBubblingEventBlock)
RCT_EXPORT_VIEW_PROPERTY(forcedOrientation, NSString*)
RCT_EXPORT_VIEW_PROPERTY(showOverlay, BOOL)

@end

File: mobile-platform/ios/NoiceStreamView/NoiceStreamViewManager.swift
//
//  NoiceStreamViewModule.swift
//  MobilePlatform
//
//  Created by Tobias Helsing on 24.9.2024.
//

import Foundation
import React
import SwiftUI

@objc(NoiceStreamViewManager)
class NoiceStreamViewManager: RCTViewManager {
  
  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  override func view() -> UIView! {
    let view = NoiceStreamView(bridge: self.bridge)
    return view
  }
  
  @objc func setOnHeaderLayoutChanged(_ view: NoiceStreamView, callback: @escaping RCTBubblingEventBlock) {
     view.onHeaderLayoutChanged = callback
   }
   
   @objc func setOnContentLayoutChanges(_ view: NoiceStreamView, callback: @escaping RCTBubblingEventBlock) {
     view.onContentLayoutChanged = callback
   }
  
}
